public/app/plugins/datasource/prometheus/components/PromQueryField.tsx

```diff
@@ -270,7 +268,7 @@ class PromQueryField extends React.PureComponent<PromQueryFieldProps, PromQueryF
 
     // Hint for big disabled lookups
     let hint: QueryHint;
-    if (lookupsDisabled) {
+    if (!datasource.lookupsDisabled && languageProvider.lookupsDisabled) {
```

Comment:
So do we want to keep both automatic disabling and manual disabling at the same time? Also where do we show this hint? I feel it may be useful to have something like that also for the manual one. Like showing `Dynamic label lookup is disabled in data source config.` so users are less confused why it does not work.

Reply:
It's not really "at the same time" but rather "possibly different configurations for multiple prometheus data sources". Pretty much in accordance with your suggestion to have a different error message.
I'll break this up and show different messages, good idea.

Reply:
But do I understand correctly that if data source has the lookupsDisabled == false it can still get disabled automatically if we see that the data source has lots of metrics? I don't have a strong opinion here just thought going for manual on/off switch we will remove the automatic one because it wasn't working properly anyway (otherwise we would not need the manual switch).

Reply:
Well, the way I figure it, the dynamic one might catch a couple of cases where people have >10K metric names but not so many that it makes grafana unusable. Because IIRC, it only gets problematic when they're in the hundreds of thousands to millions. So the automatic disabling may work for people who have something like 10K-200K metric names.